heron/packing/src/java/com/twitter/heron/packing/roundrobin/ResourceCompliantRRPacking.java

```diff
@@ -183,78 +183,43 @@ public PackingPlan pack() {
    * @return new packing plan
    */
   public PackingPlan repack(PackingPlan currentPackingPlan, Map<String, Integer> componentChanges) {
-    int adjustments = 0;
     this.numContainers = currentPackingPlan.getContainers().size();
     resetState();
 
     int additionalContainers = computeNumAdditionalContainers(componentChanges, currentPackingPlan);
     increaseNumContainers(additionalContainers);
-    LOG.info(String.format("Allocated "
-            + "%s additional containers for repack bring the number of containers to %s.",
+    LOG.info(String.format(
+        "Allocated %s additional containers for repack bring the number of containers to %s.",
         additionalContainers, this.numContainers));
-    // Get the instances using Resource Compliant Round Robin allocation
-    Optional<Map<Integer, List<InstanceId>>> resourceCompliantRRAllocation =
-        getResourceCompliantRRAllocation(currentPackingPlan, componentChanges);
 
-    while (!resourceCompliantRRAllocation.isPresent()) {
-      //The number of containers has been updated
-      if (this.numAdjustments > adjustments) {
+    int adjustments = 0;
+    while (adjustments <= this.numAdjustments) {
+      try {
+        PackingPlanBuilder planBuilder = newPackingPlanBuilder(currentPackingPlan);
+        planBuilder.updateNumContainers(numContainers);
+        planBuilder = getResourceCompliantRRAllocation(
```

Comment:
I think you're describing a scenario where a container is added and we then catch a `ResourceExceedException` and nothing has been added so we have an empty container. [We dont't have a check for that case and we should but] What currently happens is that we add a container and a number of instnaces are added to it until it's full and _then_ we get a `ResourceExceedException`, so we're consistent and the new container is full and it's time to add another.